### ISSUE TITLE
rise-local-storage is sending FILE-UPDATE as 'CURRENT' even if there was a hash mismatch on the downloaded file

### ISSUE THREAD

**Issue by santiagonoguez** [Rise-Vision/rise-launcher-electron#811]
**Describe the bug**
local-storage-module performs a hash check after download, to verify if a downloaded file is correct, here:
https://github.com/Rise-Vision/local-storage-module/blob/master/src/files/file-system.js#L122-L128

If files doesn't match it reports an error to the clients waiting on that file, from local storage log:

```
2019/09/05 11:25:42
Broadcasting FILE-ERROR to ["ws-client"] with {"filePath":"risevision-company-notifications/7fa5ee92-7deb-450b-a8d5-e5ed648c575f/template-data/06c75877-308d-4bca-a859-7928439b6beb/published/attribute-data.json","msg":"File I/O Error","detail":"Hash value does not match"}
```

When this happens, the file is not copied to the cache directory, which it's OK as it was not correctly downloaded.

The problem is that immediately after that, local storage module is reporting the file as status CURRENT:

```
2019/09/05 11:25:42
Broadcasting FILE-UPDATE to ["ws-client"] with {"filePath":"risevision-company-notifications/7fa5ee92-7deb-450b-a8d5-e5ed648c575f/template-data/06c75877-308d-4bca-a859-7928439b6beb/published/attribute-data.json","status":"CURRENT","version":"1567700740229091","ospath":"/home/santiago/rvplayer/modules/local-storage/cache/e333f90df910e81f44c66bdec57fa0e6","osurl":"file:///home/santiago/rvplayer/modules/local-storage/cache/e333f90df910e81f44c66bdec57fa0e6"}
```

Which in turn produces a couple of errors in templates because the target file doesn't exist:

```
data file read error
{"message":{"filePath":"risevision-company-notifications/7fa5ee92-7deb-450b-a8d5-e5ed648c575f/template-data/06c75877-308d-4bca-a859-7928439b6beb/published/attribute-data.json","fileUrl":"file:///home/santiago/rvplayer/modules/local-storage/cache/e333f90df910e81f44c66bdec57fa0e6","status":"CURRENT"},"stack":"Request failed: {\"isTrusted\":true} : file:///home/santiago/rvplayer/modules/local-storage/cache/e333f90df910e81f44c66bdec57fa0e6"}
```

**Reach**
Several uses may already be affected by this, if they downloaded a video, image or file but couldn't be successfully downloaded. Local storage module may keep it as CURRENT and consider it doesn't need to be downloaded again ( I'm not sure if there are existence checks to prevent this ).

**Impact**
Republishing the file usually corrects the problem, as local storage attempts again the download, which may solve the issue.

Again, I'm not sure if there are existence checks on cache files, so they are retried if not present when the display reboots. It seems not as I've seeing the problems persisting after player restarts.

**To Reproduce**
I staged this version of local-storage-module that always sends a hash mismatch:
https://github.com/Rise-Vision/local-storage-module/compare/staging/force-hash-error?expand=1
https://console.cloud.google.com/storage/browser/install-versions.risevision.com/staging/local-storage/2019.09.05.15.47/?project=avid-life-623

Add that to an overriden JSON file
```
    {
      "name": "local-storage",
      "version": "2019.09.05.15.47",
      "url": "https://storage.googleapis.com/install-versions.risevision.com/staging/local-storage/2019.09.05.15.47/local-storage-lnx-64.sh",
      "archiveSize": 0,
      "extractedSize": 0,
      "contentHash": ""
    },
```

Then create a schedule with any HTML template, such as this:
https://apps.risevision.com/schedules/details/42dc0db9-07e9-453c-9a1f-6dc38cb15f9c?cid=7fa5ee92-7deb-450b-a8d5-e5ed648c575f

Manually start the player and wait until the staged local storage is in use.

Finally, edit the presentation, save and publish so an updated version of the attribute data file is saved to published GCS.

Local storage logs will show the forced 'Hash value does not match' error, but after that it will send a FILE-UPDATE with 'CURRENT' status for the same file, even if it was not created. 

Additional errors will be logged by components that receive this CURRENT status but won't be able to use the file.

**Expected behavior**
Instead of setting FILE-UPDATE as 'CURRENT', it should be set to 'UNKNOWN', maybe this block should be reached:
https://github.com/Rise-Vision/local-storage-module/blob/master/src/files/file-controller.js#L103-L110

But it's not, maybe because the error is handled and not propagated here:
https://github.com/Rise-Vision/local-storage-module/blob/master/src/files/file.js#L78

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: I reproduced it in Linux player, but it should affect all Electron player platforms. Not sure if it also affects ChromeOS players.

**Additional context**
This may be the root cause for some of the occurrences of https://github.com/Rise-Vision/common-template/issues/83
That's maybe why some of these problems could be corrected by republishing the file.

-----

NOTE: do not add Display ID, Company ID or ScreenConnect links to this issue. Instead, manually add these details to the Trello Card. The card will be added to the [Delivery Sprint Backlog](https://trello.com/b/eDYWD5Gs/delivery) after you've submitted the issue.

**Comment by santiagonoguez** [comment]
@stulees @tejohnso @andrecardoso I can take this one as it's related to issue 83, unless anyone thinks it's better to take it instead. And if you have additional input / insights please let me know also, thanks !

**Comment by andrecardoso** [comment]
@santiagonoguez I think it makes sense you taking this one as you've already done the investigation

**Comment by santiagonoguez** [comment]
This has been rolled out at 100% now. Closing.
